File: jobs/2026-05-27__02-20-35/classify_documents_py__9DvSdTL/artifacts/user/myproject/classify.py
#!/usr/bin/env python3
import argparse
import json
import os
import sys
from pathlib import Path

from llama_cloud import LlamaCloud

RULES = [
    {
        "type": "invoice",
        "description": "documents that contain an invoice number, invoice date, bill-to section, and line items with totals.",
    },
    {
        "type": "receipt",
        "description": "short purchase receipts, typically from POS systems, with merchant, items and total, often a single page.",
    },
    {
        "type": "contract",
        "description": "multi-section legal agreement with parties, terms, and signature lines.",
    },
]


def parse_args() -> argparse.Namespace:
    parser = argparse.ArgumentParser(description="Classify a document with LlamaCloud Classifier")
    parser.add_argument("file", help="Path to the PDF document")
    return parser.parse_args()


def main() -> int:
    args = parse_args()
    file_path = Path(args.file).resolve()

    if not file_path.exists():
        print(f"File not found: {file_path}", file=sys.stderr)
        return 1

    if not os.environ.get("LLAMA_CLOUD_API_KEY"):
        print("LLAMA_CLOUD_API_KEY environment variable is not set", file=sys.stderr)
        return 1

    try:
        client = LlamaCloud()
        uploaded = client.files.create(file=str(file_path), purpose="classify")
        response = client.classifier.classify(
            file_ids=[uploaded.id],
            mode="FAST",
            rules=RULES,
        )
    except Exception as exc:
        print(f"Classifier error: {exc}", file=sys.stderr)
        return 1

    items = getattr(response, "items", None)
    if not items:
        print("Classifier returned no results", file=sys.stderr)
        return 1

    item = items[0]
    result = getattr(item, "result", None)
    if result is None:
        print("Classifier response missing result", file=sys.stderr)
        return 1

    output = {
        "file": str(file_path),
        "type": result.type,
        "confidence": result.confidence,
    }
    print(json.dumps(output))
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: jobs/2026-07-02__17-09-42/llamacloud_classify_documents_ts__ohn5k22/artifacts/user/myproject/classify.ts
import fs from "node:fs";
import path from "node:path";
import LlamaCloud from "@llamaindex/llama-cloud";

// Sample files to classify, in the order we want results to be reported.
const sampleFiles = [
  "samples/invoice.txt",
  "samples/receipt.txt",
  "samples/contract.txt",
];

// Natural-language rules for each developer-defined type.
const rules = [
  {
    type: "invoice",
    description:
      "A commercial invoice issued by a seller to a buyer. It contains an invoice number, an invoice date, a bill-to / from section, line items with quantities and unit prices, subtotal, tax, and a total amount due.",
  },
  {
    type: "receipt",
    description:
      "A point-of-sale receipt printed at the time of purchase. It shows a merchant name and address, the date and time, a short list of items purchased with individual prices, a subtotal, tax, total, and the payment method used (e.g. credit card last four digits, cash, change).",
  },
  {
    type: "contract",
    description:
      "A legal services agreement or contract between two parties. It contains recitals identifying the parties and the effective date, numbered sections describing services, term, compensation, confidentiality, and governing law, and signature blocks for both parties.",
  },
];

const outputPath = "output.log";
const runIdPath = "/logs/artifacts/run-id";

async function main() {
  // The SDK reads LLAMA_CLOUD_API_KEY from the environment by default.
  const client = new LlamaCloud();

  // Upload each sample file with purpose="classify".
  const uploaded: { basename: string; id: string }[] = [];
  for (const relativePath of sampleFiles) {
    const stream = fs.createReadStream(relativePath);
    const created = await client.files.create({
      file: stream,
      purpose: "classify",
    });
    uploaded.push({ basename: path.basename(relativePath), id: created.id });
  }

  // Run a single Classify job over all uploaded files in FAST mode.
  const results = await client.classifier.classify({
    file_ids: uploaded.map((f) => f.id),
    rules,
    mode: "FAST",
  });

  // Read the run id supplied by the parallel-run harness.
  const runId = fs.readFileSync(runIdPath, "utf8").trim();

  // items are returned in the same order as file_ids.
  const lines: string[] = [];
  lines.push(`Run ID: ${runId}`);
  for (let i = 0; i < uploaded.length; i++) {
    const { basename } = uploaded[i];
    const item = results.items[i];
    const type = item?.result?.type ?? "unknown";
    const confidence = item?.result?.confidence ?? 0;
    lines.push(`Classified: ${basename} | Type: ${type} | Confidence: ${confidence}`);
  }

  fs.writeFileSync(outputPath, lines.join("\n") + "\n", "utf8");
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
